src/coreclr/jit/importer.cpp

```diff
@@ -5372,20 +5796,125 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
         result = SimdAsHWIntrinsicInfo::lookupId(&sig, className, methodName, enclosingClassName, sizeOfVectorT);
     }
 #endif // FEATURE_HW_INTRINSICS
-    else if ((strcmp(namespaceName, "System.Runtime.CompilerServices") == 0) &&
-             (strcmp(className, "RuntimeHelpers") == 0))
+    else if (strcmp(namespaceName, "System.Runtime.CompilerServices") == 0)
     {
-        if (strcmp(methodName, "CreateSpan") == 0)
-        {
-            result = NI_System_Runtime_CompilerServices_RuntimeHelpers_CreateSpan;
-        }
-        else if (strcmp(methodName, "InitializeArray") == 0)
+        if (strcmp(className, "Unsafe") == 0)
         {
-            result = NI_System_Runtime_CompilerServices_RuntimeHelpers_InitializeArray;
+            if (strcmp(methodName, "Add") == 0)
```

Comment:
I assume the perf of all these strcmps will not be material. Otherwise you could optimize to checks on first char being < or >= 'M' for example

Reply:
This is the standard checks we have for all of these at the moment. If it ever becomes a concern, we should handle it more generally.